### PR TITLE
fix: remove compilation problems

### PR DESCRIPTION
**Pull request by juwishmaster**
I've  unused const.

Fix bugs on int always null.

i've reformated accendently a file sorry...

And send pull request.

### CODE REVIEW

File: deps/asprintf/asprintf.c [hunk 1]
```diff
@@ -7,54 +7,54 @@
...
+int asprintf(char **str, const char *fmt, ...) {
   va_list args;
 
-  // init variadic argumens
+  // init variadic arguments
   va_start(args, fmt);
 
   // format and get size
-  size = vasprintf(str, fmt, args);
+  int size = vasprintf(str, fmt, args);
```
Comment (stephenmathieson) [inline deps/asprintf/asprintf.c:23]: What OS/compiler are you using where this caused a complication _problem_?

**Review by stephenmathieson — CHANGES_REQUESTED** [review]
We cannot accept this patch.

All of these changes will be reverted if/when we update any of the packages in `deps/`. If you are having complication problems due to vendor code (files in `deps/*`), the changes will need to be made in those packages' respective repositories, then pulled into the main `clib` repository.

However, if you are unable to compile `clib` using your OS/compiler, please let us know. These changes seem very easy to get integrated upstream (into the `deps/*` packages) if they solve an _actual problem_.

File: deps/asprintf/asprintf.c (hunk 1 above)
Comment (juwishmaster) [reply deps/asprintf/asprintf.c:23]: I've used make 4.3 on archlinux system

**Comment by juwishmaster** [comment]
i don't understand why on my computer it's compile and not on your computer i'm sorry for the desagrement

File: deps/asprintf/asprintf.c (hunk 1 above)
Comment (juwishmaster) [reply deps/asprintf/asprintf.c:23]: make && sudo make install

**Comment by juwishmaster** [comment]
you should use cmake

File: deps/asprintf/asprintf.c (hunk 1 above)
Comment (stephenmathieson) [reply deps/asprintf/asprintf.c:23]: Yes, that's being done in the `Dockerfile` without error.

**Comment by stephenmathieson** [comment]
> you should use cmake

Why? The regular `Makefile` does everything we need.

**Comment by juwishmaster** [comment]
just because with cmake you can testing user operating systems before compiling

**Comment by stephenmathieson** [comment]
Can do that with [good ol' `make` too](https://github.com/clibs/clib/blob/master/Makefile#L33-L37).

To avoid more bike shedding about CMake vs make, I'm locking this thread. Please open an issue if you're unable to build clib on your system. I have not been able to reproduce, but if you can provide more information, we'd be happy to help get that sorted out.

**Pull request closed without merging**